services/motion/builtin/builtin.go

```diff
@@ -231,52 +232,262 @@ func (ms *builtIn) MoveOnGlobe(
+	goal := spatialmath.GeoPointToPose(destination, origin)
+
+	// convert GeoObstacles into GeometriesInFrame with respect to the base's starting point
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles, origin)
+
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	worldState, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+
+	// construct limits
+	straightlineDistance := goal.Point().Norm()
+	if straightlineDistance > maxTravelDistance {
+		return false, fmt.Errorf("cannot move more than %d kilometers", int(maxTravelDistance*1e-6))
+	}
+	limits := []referenceframe.Limit{
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+
+	if extra != nil {
+		if profile, ok := extra["motion_profile"]; ok {
+			motionProfile, ok := profile.(string)
+			if !ok {
+				return false, errors.New("could not interpret motion_profile field as string")
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+	}
+	ms.logger.Debugf("base limits: %v", limits)
+
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return false, resource.NewNotFoundError(componentName)
+	}
+	b, ok := baseComponent.(base.Base)
+	if !ok {
+		return false, fmt.Errorf("cannot move component of type %T because it is not a Base", baseComponent)
+	}
+
+	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, localizer, limits, kinematicsOptions)
+	if err != nil {
+		return false, err
+	}
+
+	// we take the zero position to be the start since in the frame of the localizer we will always be at its origin
+	inputMap := map[string][]referenceframe.Input{componentName.Name: make([]referenceframe.Input, len(kb.Kinematics().DoF()))}
+
+	// create a new empty framesystem which we add the kinematic base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+	kbf := kb.Kinematics()
+	if err := fs.AddFrame(kbf, fs.World()); err != nil {
+		return false, err
+	}
+
+	// TODO(RSDK-3407): this does not adequately account for geometries right now since it is a transformation after the fact.
+	// This is probably acceptable for the time being, but long term the construction of the frame system for the kinematic base should
+	// be moved under the purview of the kinematic base wrapper instead of being done here.
+	offsetFrame, err := referenceframe.NewStaticFrame("offset", movementSensorToBase.Pose())
+	if err != nil {
+		return false, err
+	}
+	if err := fs.AddFrame(offsetFrame, kbf); err != nil {
+		return false, err
+	}
+
+	// start goroutine to execute plan
+	backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+		for {
+			select {
+			case <-ctx.Done():
+				errChan <- ctx.Err()
```

Comment:
Same here; I don't think we actually need to bubble this up

Reply:
Wouldn't you want to know if its been cancelled if you were a caller?

Reply:
I don't think it does. This is an internal function, and gets cancelled in one of two situations:
1) The caller is the one cancelling it, in which case it just needs to know that this exits and returns, which the waitgroup does.
2) The caller is also being cancelled, in which case it's just trying to clean up and doesn't care about anything coming down the error channel.

Reply:
That makes sense, I'll have it return without an error in that case